cmake/project.cmake.in

```diff
@@ -234,11 +234,12 @@ macro(write_config)
             endif()
         endif()
 
-        json_write("${CMAKE_CURRENT_BINARY_DIR}/config/${plugin}.json" ${plugin_config})
```

Comment:
I do agree with the findings from @pwielders  and @sebaszm.  

The \<pluginNamePlugin\>.json file in the implementation folder is used for specifying/overriding plugins configuration and rather than specifying the interface definition. 

The interface definition may referred  in \<pluginNamePlugin\>.json as following

    "interface": {
      "$ref": "{interfacedir}/<pluginName>.json#"
    }